data/archiver/aws_s3_client.py

```diff
@@ -0,0 +1,66 @@
+import os
+import boto3
+import logging
+from data.archiver.dataclass import DataArchiverRequest, DataArchiverResult
+from data.archiver.file_transfer import FileTransfer, TransferProgress, transfer
+from data.archiver.config import AWS_S3_REGION, AWS_S3_BUCKET, AWS_ACCESS_KEY, AWS_SECRET_KEY
+
+
+class AwsS3:
+
+    def __init__(self):
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logging.INFO)
+
+    def get_files(self, req: DataArchiverRequest, res: DataArchiverResult):
+        bucket = self.new_session().resource('s3').Bucket(AWS_S3_BUCKET)
+
+        fs = []
+        for obj in bucket.objects.filter(Prefix=req.sub_uuid):
+            # skip the top-level directory
+            if obj.key == req.sub_uuid:
+                continue
+            f_name = obj.key.split('/')[1]
+            if f_name in req.files:
+                fs.append(FileTransfer(path=os.getcwd(), key=obj.key, size=obj.size))
+
+
+        def download(idx):
```

Comment:
What is this `idx`? Ths s3 id? hca id?

Reply:
Code refactored. FileProgress, FileTransfer, etc now removed.